File: Kotlin/LL_Kotlin_for_Java_Developers/src/main/java/com/aalzatea/courses/linkedinlearning/challenges/java/comparator/Smaller.java
package com.aalzatea.courses.linkedinlearning.challenges.java.comparator;

public final class Smaller {

    public static <T extends Comparable<T>> T getSmaller(T param1, T param2) {
        var result = param1.compareTo(param2);
        return result < 0 ? param1 : param2;
    }
}

File: Kotlin/LL_Kotlin_for_Java_Developers/src/main/kotlin/com/aalzatea/courses/linkedinlearning/challenges/kotlin/comparator/Smaller.kt
package com.aalzatea.courses.linkedinlearning.challenges.kotlin.comparator

class Smaller {

    companion object {
        fun <T: Comparable<T>> getSmaller(param1: T, param2: T): T {
            val result = param1.compareTo(param2)
            return if (result < 0) param1 else param2
        }
    }
}